modulemd/v2/include/modulemd-2.0/modulemd-module-stream.h

```diff
@@ -68,6 +68,9 @@ struct _ModulemdModuleStreamClass
                                  const gchar *module_name,
                                  const gchar *module_stream);
 
+  gboolean (*equals) (ModulemdModuleStream *self_1,
```

Comment:
This needs to be added at the end of the structure, just before the padding. Otherwise it breaks ABI.

And I just realized this may have happened in the other `.equals()` patches without my noticing and I need to fix that.

Reply:
The reason is that any software that was compiled against an earlier version of libmodulemd that may have been accessing this class directly needs to have the offset of the functions remain at the same place. Putting this above them means that code compiled against an earlier version trying to access `stream_class>validate()` will now be incorrectly calling `stream_class->equals()`.

Reply:
Turns out that `ModulemdComponents` and `ModulemdDefaults` both had the `.equals()` in the right place, but the padding wasn't decremented. I just pushed https://github.com/fedora-modularity/libmodulemd/commit/06611a77ac4da954d125b42f2452165c6f544879 to fix that.

Reply:
So does this mean that softwares look for linewise implementation in libmodulemd rather than syntactically checking against the function calls?

Reply:
That's an easy way to think about it, but really what happens is that the object is converted by the compiler into a contiguous memory block. Programs compiled against libmodulemd would look at an offset from the start of that block to retrieve a [32|64]-bit pointer to the function to call. If we reorder those pointers in the C file, that translates into changing the offset from the start of that memory location. So if an application was looking for a function pointer at block_start+4 which was "validate()" when this application was compiled, but now block_start+4 is returning the pointer to "equals()", the program won't behave correctly.

Reply:
Oh okay, got it! I'll make all the changes soon